### ISSUE TITLE
No file replacement popup on Linux SFTP clients

### ISSUE THREAD

**Issue by metabrixkt** [pterodactyl/panel#4739]
### Current Behavior

Replacing an existing file replaces it without any prompt on Linux file explorers (such as `thunar`, `nautilus`; they have built-in SFTP support, probably using gvfs-sftp).

### Expected Behavior

Replacing an existing file makes Linux file explorers open a popup like `This folder already contains a file "test-empty file", do you want to replace the existing file <detailed info about the existing file> with the following file? <detailed info about the new file>`. This is a behavior that I have with any OpenSSH server, but not with Pterodactyl SFTP server.

![image](https://user-images.githubusercontent.com/64396538/229128439-a8d927be-99a6-485a-b16d-b053a656d054.png)

### Steps to Reproduce

Linux required. I only tested on `thunar` (xfce4 built-in file manager) and on `nautilus` (gnome built-in file manager iirc)

1. Upload any file named `example-file` to the container
2. Create a file named `example-file` in your local folder, the contents do not matter
3. Copy this file
4. Paste the copied file to the container to the same directory where the initial `example-file` was uploaded to

On OpenSSH servers you would get a detailed prompt to replace the exisiting prompt, but on Pterodactyl SFTP server the file just gets replaced without any prompt.

In Pterodactyl Discord someone told me that it might be because wings doesn't provide a shell, but I'm just quoting this, I'm not sure about it myself as I don't know what's causing the prompt not to open. If it's because wings doesn't provide a shell, then this issue is a feature request at least for the file replacement prompt, I think?

### Panel Version

1.11.3

### Wings Version

1.11.5

### Is there an existing issue for this?

- [X] I have searched the existing issues before opening this issue.
- [X] I have provided all relevant details, including the specific game and Docker images I am using if this issue is related to running a server.
- [X] I have checked in the Discord server and believe this is a bug with the software, and not a configuration issue with my specific system.

**Comment by parkervcp** [comment]
This is likely due to those using an ssh based system and not true sftp so far as I can tell. They most likely use `scp filename host:filename` which does ask for confirmation. 

This is also true for cinnamons `nemo` file application

**Comment by metabrixkt** [comment]
Can someone remove `not confirmed` label and add `bug` label? One of the members confirmed the bug

**Comment by Sharktheone** [comment]
I'd say, this is not an issue with pterodactyl, but it's dependencies.

**Comment by metabrixkt** [comment]
> I'd say, this is not an issue with pterodactyl, but it's dependencies.

It's no longer "not confirmed" though :thinking:

**Comment by Sharktheone** [comment]
Yes, true. I guess, the `dependency` label would make more sense.

**Comment by matthewpi** [comment]
> I'd say, this is not an issue with pterodactyl, but it's dependencies.

We manage everything about the SFTP server except for the protocol itself.  All file operations hit our code, so chances are this is on our end.  My best guess without looking too far into it is that the client is expecting an `exists` error while the write path in the SFTP server is wired up to the same Write call the Panel's front-end uses which just overwrites existing files without throwing an error.

**Comment by Sharktheone** [comment]
Ah, okay. I only saw `github.com/pkg/sftp` as dependency, so i assumed that this is responsible for most of the implementation.

**Comment by metabrixkt** [comment]
_the issue purge is good, but was this fixed?_ :eyes:

**Comment by matthewpi** [comment]
> _the issue purge is good, but was this fixed?_ 👀

I don't think so, this got caught up as a "issue that is a bug but wasn't labelled" category.

**Comment by metabrixkt** [comment]
> All file operations hit our code, so chances are this is on our end.

That's correct, the SFTP handler implementation in Wings wasn't handling exclusive-create file write operations, see pterodactyl/wings#322 for more details & the fix
